src/recon.rs

```diff
@@ -263,15 +265,15 @@ impl_MergeInt!(u32, u16);
 impl_MergeInt!(u64, u32);
 impl_MergeInt!(u128, u64);
 
-#[inline]
-fn get_skip_ctx(
-    t_dim: &TxfmInfo,
+#[inline(always)]
+fn get_skip_ctx<const TX: usize>(
     bs: BlockSize,
     a: &[u8],
     l: &[u8],
     chroma: bool,
     layout: Rav1dPixelLayout,
 ) -> InRange<u8, 0, { 13 - 1 }> {
+    let t_dim = dav1d_txfm_dimension::<TX>();
```

Comment:
```suggestion
    let t_dim = const { dav1d_txfm_dimension::<TX>() };
```

Reply:
I can add that if you'd like. 

In theory it shouldn't matter, the results should be the same constant value then copied to a local whether or not it passes via the anonymous constant. 

Did you see differences in how rustc optimizes this here, or elsewhere in the project like `decode_coefs_class`? If so, maybe opening a rust issue for the const eval group could be interesting.

Reply:
I've pushed a commit to do this.

Reply:
> Did you see differences in how rustc optimizes this here, or elsewhere in the project like `decode_coefs_class`? If so, maybe opening a rust issue for the const eval group could be interesting.

No, just that the `const` here ensures that a `panic!()` in the `const fn` is assuredly optimized out, so we don't need any unsafe.

I expect this to not change codegen in practice; it's more about the guarantee.

Reply:
And it also shows intent that we really really want this to be const, which is good as well ^^.